plugins/src/webrtcsink/imp.rs

```diff
@@ -2519,3 +2590,22 @@ impl ChildProxyImpl for WebRTCSink {
         }
     }
 }
+
+impl NavigationImpl for WebRTCSink {
+    fn send_event(&self, _imp: &Self::Type, event_def: gst::Structure) {
+        let mut state = self.state.lock().unwrap();
+
+        state
+            .streams
+            .iter_mut()
+            .for_each(|(_, stream)| {
+                if stream.sink_pad.name().starts_with("video_") {
+                    gst_log!(CAT, "Navigating to: {:?}", event_def);
+                    // FIXME: Handle multi tracks.
+                    if !stream.sink_pad.push_event(gst::event::Navigation::new(event_def.clone())) {
```

Comment:
Why not

```rust
let event = gst::event::Navigation::new(event_def);
```

at the very top and then just

```rust
if !stream.sink_pad.push_event(event.clone());
```

in the loop? That way you push the same event to all pads (refcounting!) instead of creating a copy of the whole structure for every pad.

Reply:
That also implies we could create a useless event and webrtcsink doesn't seem to really handle several streams of a type yet.

Reply:
I haven't actually tested that, but what makes you say it won't handle it? For instance for congestion control I take into account the number of input video streams :)

Reply:
I have seen a few places where some more has to be done, can't recall what exactly. Not saying it is not handled at all but I could notice it wasn't properly tested :-)